### PR TITLE
[AlphaRowset] set num_segments field in rowset meta if missing

### PR DESCRIPTION
**Pull request by morningman**
the num segments should be read from rowset meta pb.
But the previous code error caused this value not to be set in some cases.
So when init the rowset meta and find that the num_segments is 0(not set),
we will try to calculate the num segments from AlphaRowsetExtraMetaPB, 
and then set the num_segments field.
This should only happen in some rowsets converted from old version.
and for all newly created rowsets, the num_segments field must be set.

ISSUE #2656

### CODE REVIEW

File: be/src/olap/rowset/rowset_meta.h [hunk 1]
```diff
@@ -326,11 +326,12 @@ class RowsetMeta {
...
     // 2. the segments_overlap() flag in rowset meta is set to NONOVERLAPPING, explicitly.
     bool is_segments_overlapping() const {
-        if (num_segments() == 0) {
-            // specially for delete version
+        if (end_version() > start_version() || segments_overlap() == NONOVERLAPPING) {
             return false;
         }
-        if (end_version() > start_version() || segments_overlap() == NONOVERLAPPING) {
+
+        // some of cumulative rowsets may have no segments.(no data, or a delete version)
```
Comment (imay) [inline be/src/olap/rowset/rowset_meta.h]: I think the cumulative rowsets will hit above if clause. So this comment is some misleading.

Comment (morningman) [reply be/src/olap/rowset/rowset_meta.h]: Changed

**Review by imay — DISMISSED** [review]
LGTM

File: be/src/olap/rowset/rowset_meta.h [hunk 2]
```diff
@@ -350,6 +352,9 @@ class RowsetMeta {
         return score;
     }
 
+    const AlphaRowsetExtraMetaPB& alpha_rowset_extra_meta_pb() {
```
Comment (imay) [inline be/src/olap/rowset/rowset_meta.h]:
```suggestion
    const AlphaRowsetExtraMetaPB& alpha_rowset_extra_meta_pb() const {
```

**Review by chaoyli — APPROVED** [review]

**Pull request merged**